### PR TITLE
Plugins Catalog: show Grafana and plugin dependencies

### LINKED ISSUE #37932: Add dependencies information on the plugin details page.

### PR DESCRIPTION
**Pull request by leventebalogh**
Fixes #37932 

**What this PR does / why we need it**:
- [x] displays the Grafana dependency for the plugin _(feature-parity)_
- [x] displays dependencies on other plugins _(feature-parity)_ 
- [x] extends the `PluginDependencies` type in `@grafana/data` to be up-to-date

**A plugin with only a Grafana dependency**
Screenshot 2021-09-10 at 8 12 14

**A plugin that has both a Grafana and other plugin dependencies**
Screenshot 2021-09-10 at 7 54 09

---

**Note to the reviewer:** I was planning to add some more test-cases for the different scenarios, but unfortunately we would probably need to massage the mocking part a bit to make these different scenarios easier to test. I am planning to create a separate issue for that.

### CODE REVIEW

**Review by mckn — APPROVED** [review]
LGTM!

File: public/app/features/plugins/admin/components/PluginDetailsHeaderDependencies.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      {/* Grafana dependency */}
+      {Boolean(grafanaDependency) && (
+        <div>
+          <Icon name="grafana" />
+          Grafana {grafanaDependency}
+        </div>
+      )}
+
+      {/* Plugin dependencies */}
+      {pluginDependencies && pluginDependencies.length > 0 && (
```
Comment (jackw) [inline public/app/features/plugins/admin/components/PluginDetailsHeaderDependencies.tsx:44]: Any reason you added a test for the grafana dependency but not the plugin deps? Is it related to [this issue](https://github.com/grafana/grafana/issues/39063) somehow?

File: public/app/plugins/datasource/graphite/state/context.tsx [hunk 2]
```diff
@@ -50,13 +50,13 @@ export const GraphiteQueryEditorContext = ({
     if (state) {
       dispatch(actions.queriesChanged(queries));
     }
-  }, [dispatch, queries]);
+  }, [dispatch, queries, state]);
```
Comment (jackw) [inline public/app/plugins/datasource/graphite/state/context.tsx:53]: I'm guessing these changes are unrelated and causing linting errors?

File: public/app/features/plugins/admin/components/PluginDetailsHeader.tsx [hunk 3]
```diff
@@ -134,5 +143,8 @@ export const getStyles = (theme: GrafanaTheme2) => {
     textUnderline: css`
       text-decoration: underline;
     `,
+    textBold: css`
```
Comment (jackw) [inline public/app/features/plugins/admin/components/PluginDetailsHeader.tsx:146]: Maybe it's my eyes but I can't see where this style is used in this file. 🤔

**Review by jackw — APPROVED** [review]
LGTM! 🚀

File: public/app/plugins/datasource/graphite/state/context.tsx (hunk 2 above)
Comment (leventebalogh) [reply public/app/plugins/datasource/graphite/state/context.tsx:53]: Yes exactly, they just showed up as typescript warnings, so I thought I will just fix them. :)

File: public/app/features/plugins/admin/components/PluginDetailsHeader.tsx (hunk 3 above)
Comment (leventebalogh) [reply public/app/features/plugins/admin/components/PluginDetailsHeader.tsx:146]: Ohh you are right, it's left here accidentally, will remove :+

File: public/app/features/plugins/admin/components/PluginDetailsHeaderDependencies.tsx (hunk 1 above)
Comment (leventebalogh) [reply public/app/features/plugins/admin/components/PluginDetailsHeaderDependencies.tsx:44]: Yes exactly, I was starting to add the test for them, but it was a bit hard to create the mocks for these scenarios with the setup that was serving us well until now. In the end I decided to move the update of the mocking (and with that adding tests for these) to a separate task (the issue that you linked). Wdyt?
